test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } elseif (isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('CE405', '0'));
+            } elseif (!isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } else {
+                print_r($error);
+                die("AE statement execution when it shouldn't!");
+            }
+        } elseif ($attestation == 'correct') {
+            if (!isEnclaveEnabled($key) and $encryptionType == 'Randomized') {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } elseif ($encryptionType == 'Deterministic') {
+                if ($comparison == '=') {
+                    print_r($error);
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } else {
+                print_r($error);
+                die("Comparison failed for correct attestation when it shouldn't have!\n");
+            }
+        } else {
+            print_r($error);
+            die("Unexpected error occurred in compareResults!\n");
+        }
+
+        return;
+    }
+
+    $AEres = $AEstmt->fetchAll(PDO::FETCH_NUM);
+    $nonAEres = $nonAEstmt->fetchAll(PDO::FETCH_NUM);
+    $AEcount = count($AEres);
+    $nonAEcount = count($nonAEres);
+
+    if ($type == 'char' or $type == 'nchar') {
+        // char and nchar may not return the same results - at this point
+        // we've verified that statement execution works so just return
+        return;
+    } elseif ($AEcount > $nonAEcount) {
+        print_r("Too many AE results!\n");
+        print_r($AEres);
+        print_r($nonAEres);
+    } elseif ($AEcount < $nonAEcount) {
+        print_r("Too many non-AE results!\n");
+        print_r($AEres);
+        print_r($nonAEres);
+    } else {
+        if ($AEcount != 0) {
+            $i = 0;
+            foreach ($AEres as $AEr) {
+                if ($AEr[0] != $nonAEres[$i][0]) {
+                    print_r("AE and non-AE results are different! For field $i, got AE result ".$AEres[$i][0]." and non-AE result ".$nonAEres[$i][0]."\n");
+                }
+                ++$i;
+            }
+        }
+    }
+}
+
+// testCompare selects based on a comparison in the WHERE clause and compares
+// the results between encrypted and non-encrypted columns, checking that the
+// results are identical
+function TestCompare($conn, $tableName, $comparisons, $dataTypes, $colNames, $thresholds, $length, $key, $encryptionType, $attestation)
+{
```

Comment:
Please rename the function to be consistent with the one in the comment.
Also, this function has many arguments, so please briefly explain what each argument is for.
You can use the function getInsertSqlComplete() in sqlsrv/MsHelper.inc  as an example